2-mandatory/6-journey-planner.js

```diff
@@ -122,9 +153,34 @@ function getLocationName() {}
   Advanced challange: try to use arrow function when invoking an array method.
 */
 function journeyPlanner(locations, transportMode) {
-  // Implement the function body
+ 
+  // SUBARRAY METHOD from above
+  // function isAccessibleByTransportMode(array, string) {
+  //   return array.some(element => element === string)
+  // }
+
+  // SUBARRAY METHOD from above
+  // function getLocationName(array) {
+  //   return array[0];
+  // }
+
+  // return locations.filter(element => isAccessibleByTransportMode(element, transportMode))
+  //                 .map(element => getLocationName(element))
+
+  return locations.filter(element => element.indexOf(transportMode) > -1)
+                  .map(element => element[0]);
```

Comment:
The reason to use the functions is to make it more readable, the names of the functions help to document